crates/node/p2p/src/net/driver.rs

```diff
@@ -57,6 +59,50 @@ impl Network {
         self.unsafe_block_signer_sender.take()
     }
 
+    /// Handles the sync request/response protocol.
+    ///
+    /// This is a mock handler that supports the `payload_by_number` protocol.
+    /// It always returns: not found (1), version (0). `<https://specs.optimism.io/protocol/rollup-node-p2p.html#payload_by_number>`
+    ///
+    /// ## Note
+    ///
+    /// This is used to ensure op-nodes are not penalizing kona-nodes for not supporting it.
+    /// This feature is being deprecated by the op-node team. Once it is fully removed from the
+    /// op-node's implementation we will remove this handler.
+    async fn sync_protocol_handler(mut sync_protocol: IncomingStreams) {
+        loop {
+            let Some((peer_id, mut inbound_stream)) = sync_protocol.next().await else {
+                warn!(target: "node::p2p::sync", "The sync protocol stream has ended");
+                return;
+            };
+
+            info!(target: "node::p2p::sync", "Received a sync request from {peer_id}, spawning a new task to handle it");
+
+            tokio::spawn(async move {
+                let mut buffer = Vec::new();
+                let Ok(bytes_received) = inbound_stream.read_to_end(&mut buffer).await else {
+                    error!(target: "node::p2p::sync", "Failed to read the sync request from {peer_id}");
+                    return;
+                };
+
+                debug!(target: "node::p2p::sync", bytes_received = bytes_received, peer_id = ?peer_id, payload = ?buffer, "Received inbound sync request");
+
+                // We return: not found (1), version (0). `<https://specs.optimism.io/protocol/rollup-node-p2p.html#payload_by_number>`
+                // Response format: <response> = <res><version><payload>
+                // No payload is returned.
+                let output = b"10";
```

Comment:
This won't return the correct response. `b"10"` is `[49, 48]`.
```suggestion
                let output = hex!("0100");
```
or just `[0x01, 0x00]`